### PR TITLE
🐛 [#4241] Fix validation of field with nested key in fieldset

### LINKED ISSUE #4241: Backend validation not applied for components inside fieldsets (with a nested key)
### Product versie / Product version

2.6.x

### Omschrijf het probleem / Describe the bug

Only when the key of the component inside the fieldset is `fieldset_key.component_key`, backend validation will not be applied.

Regression test:

```python
def test(self):
    component: FieldsetComponent = {
        "type": "fieldset",
        "key": "fieldset",
        "label": "A field set",
        "components": [
            {
                "type": "textfield",
                "key": "fieldset.textfield",
                "label": "Text field",
                "validate": {"required": True},
            },
        ],
        "hidden": False,
    }

    is_valid, errors = validate_formio_data(component, {"fieldset": {"textfield": ""}})

    self.assertFalse(is_valid)
```

### Stappen om te reproduceren / Steps to reproduce

_No response_

### Verwacht gedrag / Expected behavior

_No response_

### Screen resolution

None

### Device

None

### OS

None

### Browser

_No response_

### PR DESCRIPTION
**Pull request by stevenbal**
Closes #4241 

**Changes**

* Fix validation of field with nested key in fieldset

**Checklist**

Check off the items that are completed or not relevant.

- Impact on features

  - [x] Checked copying a form
  - [x] Checked import/export of a form
  - [x] Config checks in the configuration overview admin page
  - [x] Problem detection in the admin email digest is handled

- Release management

  - [x] I have labelled the PR as "needs-backport" accordingly

- I have updated the translations assets (you do NOT need to provide translations)

  - [x] Ran `./bin/makemessages_js.sh.sh`
  - [x] Ran `./bin/compilemessages_js.sh`

- Commit hygiene

  - [x] Commit messages refer to the relevant Github issue
  - [x] Commit messages explain the "why" of change, not the how

### CODE REVIEW

File: src/openforms/formio/rendering/nodes.py [hunk 1]
```diff
@@ -230,7 +230,10 @@ def __iter__(self) -> Iterator["ComponentNode"]:
             return
 
         # in export mode, only emit if the component is not a layout component
-        if self.mode != RenderModes.export or not is_layout_component(self.component):
+        if self.mode != RenderModes.export or (
+            not is_layout_component(self.component)
+            and self.component["type"] != "editgrid"
```
Comment (stevenbal) [inline src/openforms/formio/rendering/nodes.py:235]: @sergei-maertens I do have to add this check here explicitly after I added the `editgrid` check to `is_layout_component`

File: src/openforms/formio/utils.py [hunk 2]
```diff
@@ -159,6 +159,9 @@ def get_readable_path_from_configuration_path(
 def is_layout_component(component: Component) -> bool:
     # Adapted from isLayoutComponent util function in Formio
     # https://github.com/formio/formio.js/blob/4.13.x/src/utils/formUtils.js#L25
+    if component["type"] == "editgrid":
+        return False
```
Comment (sergei-maertens) [inline src/openforms/formio/utils.py:163]: Can you add a FIXME that this isn't the neatest way to handle this? :grimacing:

File: src/openforms/formio/rendering/nodes.py (hunk 1 above)
Comment (sergei-maertens) [reply src/openforms/formio/rendering/nodes.py:235]: yeah we shoved editgrid in there in a weird way, I'm glad it's not worse than this

**Comment by stevenbal** [comment]
@sergei-maertens I'm not sure why codecov is failing here, if I put a breakpoint after line 99 locally, I see that this case is executed in the `openforms.formio` tests: https://app.codecov.io/gh/open-formulieren/open-forms/pull/4329/blob/src/openforms/formio/serializers.py

File: src/openforms/formio/utils.py (hunk 2 above)
Comment (stevenbal) [reply src/openforms/formio/utils.py:164]: added it

**Comment by sergei-maertens** [comment]
Backported to 2.6 with 4bf6f06bb2516502c79bfd1442e55423844d9a3c

**Pull request merged**